cloud/scope/powervs_cluster.go

```diff
@@ -1627,71 +1651,66 @@ func (s *PowerVSClusterScope) ReconcileTransitGateway() (bool, error) {
 		if err != nil {
 			return false, err
 		}
-		requeue, err := s.checkTransitGateway(tg.ID)
+		requeue, _, _, err := s.checkAndUpdateTransitGateway(tg, false)
```

Comment:
In a case where TG pre-exists and no connections exist, if we pass `false` here, it won't create any connections right?

Reply:
Modified the logic to handle the existing resource passed via id in second block and taken care there to update the status. Ptal!